### PR TITLE
add iostream api to write read file stream

### PR DESCRIPTION
**Pull request by diegodox**
This PR provide api using `iostream` to write code like below.

```c++
    const auto data = std::vector<int>{1, 2, 3, 4, 5, 8, 3, 2, 1, 0, 3, 7};
    // encode and write file
    {
        std::ofstream ofs("./test.dat", std::ios::binary);
        rangecoder::RangeEncoder enc;
        for (int i = 0; i < data.size(); i++)
        {
            enc.encode(rangecoder::UniformDistribution<16>(), data[i]);
        }
        ofs << enc;
    }
    {
        std::ifstream ifs("./test.dat", std::ios::binary);
        rangecoder::RangeDecoder dec;
        ifs >> dec;
        auto decoded = std::vector<int>();
        for (int i = 0; i < data.size(); i++)
        {
            decoded.push_back(dec.decode(rangecoder::UniformDistribution<16>()));
        }
        EXPECT_EQ(decoded, data);
    }
```

### CODE REVIEW

**Comment by diegodox** [comment]
@hiroki-kojima Would you review this?

**Comment by hiroki-kojima** [comment]
@diegodox LGTM :eyes:

File: rangecoder.h [hunk 1]
```diff
@@ -227,6 +227,17 @@ namespace rangecoder
             return m_bytes;
         }
 
+        friend std::ostream &operator<<(std::ostream &os, RangeEncoder &re)
+        {
+            auto data = re.finish();
```
Comment (hiroki-kojima) [inline rangecoder.h:232]: const ?

Comment (diegodox) [reply rangecoder.h:232]: yes, const

**Comment by diegodox** [comment]
@hiroki-kojima `const` added, anything else I miss?

**Comment by hiroki-kojima** [comment]
@diegodox LGTM

**Pull request merged**